### ISSUE TITLE
Can we use the operator without the plugin

### ISSUE THREAD

**Issue by harishmurthy** [kfsoftware/hlf-operator#54]
Is it possible to use the hlf operator without the plugin, say using CRD and helm charts? We have a requirement where the deployment needs to happen through helm charts, not through CLI.

Much appreciate your help in this regard.

Thanks

**Comment by dviejokfs** [comment]
Hi @harishmurthy 

Take a look at this project using Go with the Kuberntes client you can create resources programmatically.

https://github.com/kfsoftware/hlf-operator-ui/tree/main/api

**Comment by nkaramolegos** [comment]
Can you give us an example on that? For example how the peer creation is taking place using the api

**Comment by nkaramolegos** [comment]
@dviejokfs As I can see [here](https://github.com/kfsoftware/hlf-operator-ui/blob/main/api/gql/resolvers/mutation.go) the functionalities for the deployment are not implemented. If I don't want to use bash with the `hlf plugin` to deploy (and maintain -i.e add an orderer node after the initial deployment-) the network what is the best way to make the job done. Should I make a custom Go-lang application where I will call the functions that hlf plugin uses? For example for creating a peer to call [this](https://github.com/kfsoftware/hlf-operator/blob/e82d59bf3eaf69be2586d0191e4e4a4caef75868/kubectl-hlf/cmd/peer/create.go#L55) function with the appropriate arguments?

**Comment by dviejokfs** [comment]
@nkaramolegos Yes, the functions you need to call are the same but instead of doing it within the Kubectl plugin, you can do it within your API.

The mutations are not implemented, but the queries are. It's almost the same are the queries but changing the operation, instead of `List`, you need to invoke the function `Create` or `Update`:
https://github.com/kfsoftware/hlf-operator-ui/blob/main/api/gql/resolvers/query.go#L23

**Comment by nkaramolegos** [comment]
So I will go to this [function](https://github.com/kfsoftware/hlf-operator-ui/blob/e2aacf1d4559161364276106bbd8dfeed87cb46c/api/gql/resolvers/mutation.go#L8) and I will add this [code](https://github.com/kfsoftware/hlf-operator/blob/e82d59bf3eaf69be2586d0191e4e4a4caef75868/kubectl-hlf/cmd/peer/create.go#L241) after creating the appropriate options and parameters. Correct?

**Comment by nkaramolegos** [comment]
Can you please provide some details on that.

**Comment by dviejokfs** [comment]
Yes @nkaramolegos , exactly, that's the way to go, can you share some code publicly for me to review it? If that's possible that would accelerate the process so much.

Thank you

**Comment by nkaramolegos** [comment]
I will try to implement that. I am new to Go and things looks a bit complicated. What I have to compile in order to call the function `CreatePeer` from my code

**Comment by nkaramolegos** [comment]
Excuse my ignorance in how things runs in Go but I have ran inside the api folder `go build -o api main.go  `. After this, how can I call [this](https://github.com/kfsoftware/hlf-operator-ui/blob/e2aacf1d4559161364276106bbd8dfeed87cb46c/api/gql/resolvers/mutation.go#L8) function to see the panic message. Is there a test script with the appropriate imports and calls with the corresponding arguments?

**Comment by nkaramolegos** [comment]
Hello @dviejokfs If you give some guidelines or an example on how to start, compile and use the api I may help you implement it. It is something that I really need for my project

**Comment by dviejokfs** [comment]
Yes — the operator works entirely through CRDs and Helm charts without requiring the kubectl plugin. The plugin (`kubectl-hlf`) is a convenience CLI but all operations can be done by applying CRD manifests directly:

- `FabricCA`, `FabricPeer`, `FabricOrderingService` for components
- `FabricMainChannel`, `FabricFollowerChannel` for channels
- `FabricNetworkConfig` for connection profiles

The Helm chart at `chart/hlf-operator/` deploys the operator itself. Component configuration is all done via CRDs.

Closing as this has been answered multiple times in the comments and is well-supported.
